cozy/view_model/storages_view_model.py

```diff
@@ -0,0 +1,120 @@
+import logging
+from threading import Thread
+
+from peewee import SqliteDatabase
+
+from cozy.application_settings import ApplicationSettings
+from cozy.architecture.event_sender import EventSender
+from cozy.architecture.observable import Observable
+from cozy.control.filesystem_monitor import FilesystemMonitor
+from cozy.ext import inject
+from cozy.media.importer import Importer
+from cozy.model.library import Library
+from cozy.model.settings import Settings
+from cozy.model.storage import Storage
+
+log = logging.getLogger("storages_view_model")
+
+
+class StoragesViewModel(Observable, EventSender):
+    _library: Library = inject.attr(Library)
+    _importer: Importer = inject.attr(Importer)
+    _model: Settings = inject.attr(Settings)
+    _app_settings: ApplicationSettings = inject.attr(ApplicationSettings)
+    _db = inject.attr(SqliteDatabase)
+    _fs_monitor = inject.attr(FilesystemMonitor)
+
+    def __init__(self) -> None:
+        super().__init__()
+        super(Observable, self).__init__()
+
+        self._selected_storage = None
+
+    def _scan_new_storage(self, model: Storage) -> None:
+        self.emit_event("storage-added", model)
+        log.info("New audiobook location added. Starting import scan.")
+        Thread(target=self._importer.scan, name="ImportThread").start()
```

Comment:
The thread (same with `RebaseStorageLocationThread`) is not explicitly marked as daemonic, so IIUC [it will inherit](https://docs.python.org/3/library/threading.html#threading.Thread) the setting from the current (main) thread, meaning that it will prevent Cozy from exiting until the `self._importer.scan` method finishes. Is this intended?

Reply:
Yup. This actually cause some weird freezes for me wen importing a large library.
Will fix. Maybe there should also be a cancel button to cancel the operation.

Reply:
Hmm. It looks like some extra work is needed to make this thread safe to be demonic (otherwise the database may corrupt), so I won't do it in this PR.